ibis/expr/operations/sortkeys.py

```diff
@@ -2,21 +2,59 @@
 
 from __future__ import annotations
 
+from typing import TypeVar
+
 from public import public
 
+import ibis.expr.datashape as ds
+import ibis.expr.datatypes as dt
 import ibis.expr.rules as rlz
 from ibis.expr.operations.core import Value
 
+# TODO(kszucs): move the content of this file to generic.py
+
+T = TypeVar("T", bound=dt.DataType)
+S = TypeVar("S", bound=ds.DataShape)
+
+
+_is_ascending = {
+    "asc": True,
+    "ascending": True,
+    "desc": False,
+    "descending": False,
+    0: False,
+    1: True,
+    False: False,
+    True: True,
+}
 
+
+# TODO(kszucs): consider to limit its shape to Columnar, we could treat random()
```

Comment:
`random()` is funky in that it effectively takes on whatever shape is dictated by the query. I don't think there's a sane way to model that in the operation itself, since it depends on the caller's context.

I don't think we can simply stipulate that it is columnar without any other modifiers to that statement, because it's a valid scalar operation.